source/joker.c

```diff
@@ -248,68 +253,77 @@ void joker_object_shake(JokerObject *joker_object, mm_word sound_id)
-        *money += joker_effect.money;
-        // TODO: Retrigger
+        return false;
+    }
 
-        const int joker_score_display_offset_px = (MAX_CARD_SCORE_STR_LEN + 1)*TTE_CHAR_SIZE;
-        // + 1 For space
+    JokerEffect joker_effect = joker_get_score_effect(joker_object->joker, scored_card, joker_event);
 
-        int cursorPosX = fx2int(joker_object->sprite_object->x) + 8; // Offset of 16 pixels to center the text on the card
-        if (joker_effect.chips > 0)
-        {
-            char score_buffer[INT_MAX_DIGITS + 2]; // For '+' and null terminator
-            tte_set_pos(cursorPosX, JOKER_SCORE_TEXT_Y);
-            tte_set_special(0xD000); // Blue
-            snprintf(score_buffer, sizeof(score_buffer), "+%d", joker_effect.chips);
-            tte_write(score_buffer);
-            cursorPosX += joker_score_display_offset_px;
-        }
-        if (joker_effect.mult > 0)
-        {
-            char score_buffer[INT_MAX_DIGITS + 2];
-            tte_set_pos(cursorPosX, JOKER_SCORE_TEXT_Y);
-            tte_set_special(0xE000); // Red
-            snprintf(score_buffer, sizeof(score_buffer), "+%d", joker_effect.mult);
-            tte_write(score_buffer);
-            cursorPosX += joker_score_display_offset_px;
-        }
-        if (joker_effect.xmult > 0)
-        {
-            char score_buffer[INT_MAX_DIGITS + 2];
-            tte_set_pos(cursorPosX, JOKER_SCORE_TEXT_Y);
-            tte_set_special(0xE000); // Red
-            snprintf(score_buffer, sizeof(score_buffer), "X%d", joker_effect.xmult);
-            tte_write(score_buffer);
-            cursorPosX += joker_score_display_offset_px;
-        }
-        if (joker_effect.money > 0)
-        {
-            char score_buffer[INT_MAX_DIGITS + 2];
-            tte_set_pos(cursorPosX, JOKER_SCORE_TEXT_Y);
-            tte_set_special(0xC000); // Yellow
-            snprintf(score_buffer, sizeof(score_buffer), "+%d", joker_effect.money);
-            tte_write(score_buffer);
-            cursorPosX += joker_score_display_offset_px;
-        }
+    if (memcmp(&joker_effect, &(JokerEffect){0}, sizeof(JokerEffect)) == 0)
+    {
+        return false;
+    }
 
-        joker_object->joker->processed = true; // Mark the joker as processed
-        joker_object_shake(joker_object, SFX_CARD_SELECT); // TODO: Add a sound effect for scoring the joker
+    *chips    += joker_effect.chips;
+    *mult     += joker_effect.mult;
+    *mult     *= joker_effect.xmult > 0 ? joker_effect.xmult : 1; // if xmult is zero, DO NOT multiply by it
+    *money    += joker_effect.money;
+    *retrigger = joker_effect.retrigger;
+    // joker_effect.message will have been set if the Joker had anything custom to say
 
-        return true;
+    int cursorPosX = fx2int(joker_object->sprite_object->x) + 8; // Offset of 16 pixels to center the text on the card
+    if (joker_effect.chips > 0)
+    {
+        char score_buffer[INT_MAX_DIGITS + 2]; // For '+' and null terminator
```

Comment:
nitpick. This is a style choice only, but I could only declare `score_buffer` once and just rewrite to it.

Reply:
either way, all good :sunglasses: